src/window/file_dialog.c

```diff
@@ -193,13 +204,15 @@ static void draw_foreground(void)
         if (data.focus_button_id == i + 1) {
             font = FONT_NORMAL_WHITE;
         }
-        encoding_from_utf8(data.file_list->files[scrollbar.scroll_position + i], file, FILE_NAME_MAX);
+        encoding_from_utf8(data.file_list->files[scrollbar.scroll_position + i].name, file, FILE_NAME_MAX);
         //file_remove_extension(file);
         text_ellipsize(file, font, MAX_FILE_WINDOW_TEXT_WIDTH);
         text_draw(file, 160, 130 + 16 * i, font, 0);
     }
 
-    image_buttons_draw(0, 0, image_buttons, 2);
+    image_buttons_draw(0, 0, ok_cancel_buttons, 2);
```

Comment:
You could keep the `image_buttons` then use `image_buttons_draw(0, 0, image_buttons, 2 + platform_file_manager_has_stat()`) (since that function shouldn't return a `bool` anyway but an `int`, either a `0` or `1`.